connection/auth.js

```diff
@@ -192,6 +196,63 @@ class AccessTokenAuthenticator {
   };
 }
 
+export class AuthClientCredentials {
+  constructor(creds) {
+    this.clientSecret = creds.clientSecret;
+  }
+}
+
+class ClientCredentialsAuthenticator {
+  constructor(http, creds, config) {
+    this.http = http;
+    this.creds = creds;
+    this.openidConfig = config;
+  }
+
+  refresh = () => {
+    this.validateOpenidConfig();
+    return this.requestAccessToken()
+      .then(tokenResp => {
+        return {
+          accessToken: tokenResp.access_token,
+          expiresAt: calcExpirationEpoch(tokenResp.expires_in),
+          refreshToken: tokenResp.refresh_token
```

Comment:
CC usually does not include a refresh token. Keycloak(==WCS) does, but the spec says to not send one and azure and okta do not.
The reasoning is that the client secret is not a real secret and so the clients are allowed to save it 🙄 So you need to save the credentials here and then resuse them to get a new token